### PR TITLE
Fix in the MPA stub building code

### PR DESCRIPTION
**Pull request by sviret**
#### PR description:

This PR implements a fix in the stub building code of the L1Trigger/TrackTrigger package. Only one file is affected (TTStubBuilder.cc), and one line in this file. The PS modules can build up to 2 stubs per seed. It was set to one until now. So this is a very minor change (1->2)

#### PR validation:

The code change was tested with a private CMSSW version. As expected a small increase of the stub rate was observed in the PS module (less than 5% in any cases @ PU200).

### CODE REVIEW

**Comment by srimanob** [comment]
Please test

**Comment by srimanob** [comment]
Hi @sviret 

Could you please confirm if all changes see in the Phase-2 workflow (SiOuterTracker/Tracks, /Stubs) are expected?
https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_12_0_X_2021-06-22-1100+198353/43600/validateJR.html

I assume yes, but better to x-check with you first. Thanks.

**Comment by sviret** [comment]
Hi @srimanob,

Those changes are indeed expected. The stub rate in the PS modules is increased by few %, this is what is detected by the DQM.

**Comment by cecilecaillol** [comment]
+l1

**Comment by srimanob** [comment]
+Upgrade

File: L1Trigger/TrackTrigger/plugins/TTStubBuilder.cc [hunk 1]
```diff
@@ -210,7 +210,7 @@ void TTStubBuilder<Ref_Phase2TrackerDigi_>::produce(edm::Event& iEvent, const ed
 
       /// Here tempOutput stores all the stubs from this lower cluster
       /// Check if there is need to store only one (if only one already, skip this step)
```
Comment (qliphy) [inline L1Trigger/TrackTrigger/plugins/TTStubBuilder.cc:212]: Should we also update the commented line accordingly?

Comment (sviret) [reply L1Trigger/TrackTrigger/plugins/TTStubBuilder.cc:212]: @qliphy thanks for the comment which actually pointed me toward a small bug in the code. I will repush a version with comments and the fix (which is just one line the loop). So please wait for the update before pushing the PR

Comment (sviret) [reply L1Trigger/TrackTrigger/plugins/TTStubBuilder.cc:212]: I'm currently running some quick checks but the impact should be negligible. The case where there is more than 2 stubs formed from one seed in a PS module was not accounted correctly (only 1 stub was kept). This should be very rare anyway, but better to implement the exact behavior

**Comment by sviret** [comment]
I made the correction and added some comments, as suggested. As expected impact on total stub rates is negligible, but now the code represents the correct behavior

**Comment by srimanob** [comment]
Please test

**Comment by sviret** [comment]
Error seems to be the following:

2021/06/23 17:45:05 ERROR failed to parse X509 proxy: crypto/tls: failed to parse key: asn1: syntax error: sequence truncated

TMO this has nothing to do with the PR
@srimanob I suspect the solution is to just relaunch the test, no?

**Comment by qliphy** [comment]
please test

**Comment by cecilecaillol** [comment]
+l1

**Comment by srimanob** [comment]
+Upgrade

- Change in Phase-2 wfs as expected.
- The total failure increases due to the issue from the input files used in workflow 140.53 (2011 HI).

**Comment by qliphy** [comment]
+1

**Pull request merged**